tests/foreman/ui_airgun/test_bookmark.py

```diff
@@ -0,0 +1,130 @@
+import random
+
+from airgun.session import Session
+from fauxfactory import gen_string
+from nailgun import entities
+
+from robottelo.constants import BOOKMARK_ENTITIES
+from robottelo.decorators import bz_bug_is_open, fixture, rm_bug_is_open, tier2
+
+
+@fixture(scope='module')
+def module_org():
+    return entities.Organization().create()
+
+
+@fixture(scope='module')
+def custom_user(module_org):
+    """Custom user with viewer role for tests validating 'public' bookmark
+    option.
+    """
+    viewer_role = entities.Role().search(query={'search': 'name="Viewer"'})[0]
+    bm_role = entities.Role().create()
+    entities.Filter(
+        permission=entities.Permission(
+            resource_type='Bookmark').search(
+            filters={'name': 'view_bookmarks'}),
+        role=bm_role,
+    ).create()
+    custom_password = gen_string('alphanumeric')
+    custom_user = entities.User(
+        default_organization=module_org,
+        organization=[module_org],
+        role=[viewer_role, bm_role],
+        password=custom_password,
+    ).create()
+    custom_user.password = custom_password
+    return custom_user
+
+
+@fixture(scope='module')
+def ui_entities(module_org):
+    """Collects the list of all applicable UI entities for testing and does all
+    required preconditions.
+    """
+    ui_entities = []
+    for entity in BOOKMARK_ENTITIES:
+        # Skip the entities, which can't be tested ATM (not implemented in
+        # airgun or have open BZs)
+        skip = entity.get('skip_for_ui')
+        if isinstance(skip, tuple):
+            if (skip[0] == 'bugzilla' and bz_bug_is_open(skip[1])
```

Comment:
prefer to unpack first
```python
bug_type, bug_number = skip
```

Reply:
Ok, but why? Just curious :) Can't see any real win in that extra operation.